Src/DependencyCollector/Shared/Implementation/HttpHandlerDiagnosticListener.cs

```diff
@@ -0,0 +1,361 @@
+        {
+            this.Initialize();
+            return base.Subscribe(observer, isEnabled);
+        }
+
+        /// <summary>
+        /// Overriding base class implementation just to give us a chance to initialize.  
+        /// </summary>
+        public override IDisposable Subscribe(IObserver<KeyValuePair<string, object>> observer, Func<string, object, object, bool> isEnabled)
+        {
+            this.Initialize();
+            return base.Subscribe(observer, isEnabled);
+        }
+
+#region private methods
+
+        private static void PrepareReflectionObjects()
+        {
+            // At any point, if the operation failed, it should just throw, including NullReferenceException.
+            // The caller should catch all exceptions and swallow.
+
+            // First step: Get all the reflection objects we will ever need.
+            Assembly systemNetHttpAssembly = typeof(ServicePoint).Assembly;
+            connectionGroupListField = typeof(ServicePoint).GetField("m_ConnectionGroupList", BindingFlags.Instance | BindingFlags.NonPublic);
+            connectionGroupType = systemNetHttpAssembly?.GetType("System.Net.ConnectionGroup");
+            connectionListField = connectionGroupType?.GetField("m_ConnectionList", BindingFlags.Instance | BindingFlags.NonPublic);
+            connectionType = systemNetHttpAssembly?.GetType("System.Net.Connection");
+            writeListField = connectionType?.GetField("m_WriteList", BindingFlags.Instance | BindingFlags.NonPublic);
+
+            // Second step: Generate an accessor for HttpWebRequest._HttpResponse
+            FieldInfo field = typeof(HttpWebRequest).GetField("_HttpResponse", BindingFlags.NonPublic | BindingFlags.Instance);
+
+            string methodName = field?.ReflectedType.FullName + ".get_" + field.Name;
+            if (!string.IsNullOrEmpty(methodName))
+            {
+                DynamicMethod getterMethod = new DynamicMethod(methodName, typeof(HttpWebResponse), new Type[] { typeof(HttpWebRequest) }, true);
+                ILGenerator generator = getterMethod.GetILGenerator();
+                generator.Emit(OpCodes.Ldarg_0);
+                generator.Emit(OpCodes.Ldfld, field);
+                generator.Emit(OpCodes.Ret);
+                httpResponseAccessor = (Func<HttpWebRequest, HttpWebResponse>)getterMethod.CreateDelegate(typeof(Func<HttpWebRequest, HttpWebResponse>));
+            }
+
+            // Double checking to make sure we have all the pieces initialized
+            if (connectionGroupListField == null ||
+                connectionGroupType == null ||
+                connectionListField == null ||
+                connectionType == null ||
+                writeListField == null ||
+                httpResponseAccessor == null)
+            {
+                throw new InvalidOperationException("Unable to initialize all required reflection objects");
```

Comment:
Will take your suggestion about changing these functions to return bool, so these strings should go away.